Sources/AblyChat/ChatAPI.swift

```diff
@@ -0,0 +1,232 @@
+            return
+        }
+        
+        guard let paginatedResponse = callback.0, paginatedResponse.statusCode == 200 else {
+            continuation.resume(throwing: ARTErrorInfo.createUnknownError())
+            return
+        }
+        
+        do {
+            let jsonData = try JSONSerialization.data(withJSONObject: paginatedResponse.items)
+            let decodedData = try JSONDecoder().decode([RES].self, from: jsonData)
+            let result = paginatedResponse.toPaginatedResult(items: decodedData)
+            continuation.resume(returning: result)
+        } catch {
+            continuation.resume(throwing: error)
+        }
+    }
+}
+
+extension ARTHTTPPaginatedResponse {
+    func toPaginatedResult<T: Codable & Sendable>(items: [T]) -> PaginatedResultWrapper<T>{
+        
+        return PaginatedResultWrapper(paginatedResponse: self, items: items)
+    }
+}
+
+public final class PaginatedResultWrapper<T: Codable & Sendable>: PaginatedResult {
+    public typealias T = T
+    
+    public let items: [T]
+    public let hasNext: Bool
+    public let isLast: Bool
+    private let paginatedResponse: ARTHTTPPaginatedResponse
+    
+    public init(paginatedResponse: ARTHTTPPaginatedResponse, items: [T]) {
+        self.items = items
+        self.hasNext = paginatedResponse.hasNext
+        self.isLast = paginatedResponse.isLast
+        self.paginatedResponse = paginatedResponse
+    }
+    
+    /// Asynchronously fetch the next page if available
+    public var next: (any PaginatedResult<T>)? {
+        get async throws {
+            if #available(macOS 13.0.0, *) {
+                return try await withCheckedThrowingContinuation { continuation in
+                    paginatedResponse.next { paginatedResponse, error in
+                        ARTHTTPPaginatedCallbackWrapper(callback: (paginatedResponse, error)).handleResponse(continuation: continuation)
+                    }
+                }
+            } else {
+                return nil
+            }
+        }
+    }
+    
+    /// Asynchronously fetch the first page
+    public var first: any PaginatedResult<T> {
+        get async throws {
+            if #available(macOS 13.0.0, *) {
+                return try await withCheckedThrowingContinuation { continuation in
+                    paginatedResponse.first { paginatedResponse, error in
+                        ARTHTTPPaginatedCallbackWrapper(callback: (paginatedResponse, error)).handleResponse(continuation: continuation)
+                    }
+                }
+            } else {
+                throw fatalError()
+            }
+        }
+    }
+    
+    /// Asynchronously fetch the current page
+    public var current: any PaginatedResult<T> {
+        get async throws {
+            return try await withCheckedThrowingContinuation { continuation in
+                continuation.resume(returning: self)
+            }
+        }
+    }
+}
+
+// TODO: Encode these instead
```

Comment:
Unless this is a TODO that you intend to resolve inside this PR, it should have a GitHub issue so that we remember to return to it.

Reply:
thought against it... current implementation is fine until I handle my other todo on mapping to `Date` types correctly